src/edge/acrManager.ts

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+"use strict";
+import { ContainerRegistryManagementClient } from "azure-arm-containerregistry";
+import { Registry, RegistryListCredentialsResult } from "azure-arm-containerregistry/lib/models";
+import * as request from "request-promise";
+import * as vscode from "vscode";
+import { Executor } from "../common/executor";
+import { UserCancelledError } from "../common/UserCancelledError";
+import { Utility } from "../common/utility";
+import { AcrRegistryQuickPickItem } from "../container/models/AcrRegistryQuickPickItem";
+import { AzureAccount, AzureSession, AzureSubscription } from "../typings/azure-account.api";
+
+export class AcrManager {
+    private readonly azureAccount: AzureAccount;
+    private refreshTokenArc: string;
+
+    constructor() {
+        this.azureAccount = vscode.extensions.getExtension<AzureAccount>("ms-vscode.azure-account")!.exports;
+    }
+
+    public async selectAcrImage(): Promise<string> {
+        const acrRegistryItem: AcrRegistryQuickPickItem = await this.selectAcrRegistry();
+        if (acrRegistryItem === undefined) {
+            throw new UserCancelledError();
+        }
+
+        const registryUrl: string = acrRegistryItem.registry.loginServer;
+
+        const acrRepoItem: vscode.QuickPickItem = await this.selectAcrRepo(registryUrl, acrRegistryItem.azureSubscription.session);
+        if (acrRepoItem === undefined) {
+            throw new UserCancelledError();
+        }
+
+        const acrTagItem: vscode.QuickPickItem = await this.selectAcrTag(registryUrl, acrRepoItem.label);
+        if (acrTagItem === undefined) {
+            throw new UserCancelledError();
+        }
+
+        this.loginToAcr(acrRegistryItem.registry, acrRegistryItem.azureSubscription);
+
+        return acrTagItem.description;
+    }
+
+    private async selectAcrRegistry(): Promise<AcrRegistryQuickPickItem> {
+        if (!(await this.azureAccount.waitForLogin())) {
+            await vscode.commands.executeCommand("azure-account.askForLogin");
```

Comment:
What if user cancel for "azure-account.askForLogin"?  You could refer to https://github.com/Microsoft/vscode-azure-iot-toolkit/blob/ed583eaebcd336a94ed2a97e740c5302cd6f2b4a/src/iotHubResourceExplorer.ts#L173

Reply:
Fixed. Thanks.